### ISSUE TITLE
Indent rule: unable to enforce indentation of switch cases [$15]

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#1797]
The following code results in having no errors:

`eslint --reset --no-eslintrc --rule "{indent: 2}" test.js`
```js
switch (a) {
case '1':
b();
break;
default:
c();
break;
}
```

If I enable the `indentSwitchCase` option I get the following errors:
`eslint --reset --no-eslintrc --rule "{indent: [2, 4, {indentSwitchCase: true}]}" test.js`

```sh
test.js
  3:4  error  Expected indentation of 4 characters  indent
  6:4  error  Expected indentation of 4 characters  indent
```

I would expect the following errors:
```sh
test.js
  2:4  error  Expected indentation of 4 characters  indent
  3:8  error  Expected indentation of 8 characters  indent
  4:8  error  Expected indentation of 8 characters  indent
  5:4  error  Expected indentation of 4 characters  indent
  6:8  error  Expected indentation of 8 characters  indent
  7:8  error  Expected indentation of 8 characters  indent
```

It seems like the `indentSwitchCase` option enables the indentation check only of the body of a SwitchCase statement. The Indentation of the whole SwitchCase statement is recognized by the rule itself and only validated to be consistet within an single switch statement. There is no option to confiugre a consistent identation for SwitchCase statements of a whole file.

The other strange behavior with this option is, when I explecitly set the value to `false` it doesn’t forbid the identation for a SwitchCase body, it just ignores those lines.

**Comment by t128** [comment]
i have a strange `"indent": [2, "tab", {indentSwitchCase: true}]`behavior too.

my code:
```
switch(test) {
	case '2':
		test(24)
		//falls through

	case '3':
		test(42)
		break
}
```

the error for `"indent": [2, "tab", {indentSwitchCase: true}]`:
```
  23:2  error  Expected indentation of 2 characters  indent
  24:3  error  Expected indentation of 3 characters  indent
  25:3  error  Expected indentation of 3 characters  indent
```

so i have to change the code to:
```
switch(test) {
	case '2':
		test(24)
		//falls through

		case '3':
			test(42)
			break
}
```

**Comment by doberkofler** [comment]
@t128 I'm also using the indent rule in an physical tab environment and cannot reproduce your problem. The following code example shown no errors in eslint 0.16.2 when using the tab character to indent by one or two levels.

```
/*eslint indent: [2, "tab", {indentSwitchCase: true}]*/

var test = "1";
switch(test) {
	case "2": // indented by 1 tab character
		test(24); // indented by 2 tab characters
		// falls through
	case "3":
		test(42);
		break;
}
```

**Comment by gyandeeps** [comment]
@lo1tuma I started working on it.
Based on your example on top, you expected line 4 to have an indention of 8 characters.
Further looking into the code I found out that it considers `break` statement to along the same column as `case` statement.
This is weird for me also. As I also expect the indentation as you expected for break statement.
Do you think the rule should be changed to accommodate this scenario also.

Any input @nzakas ?
FYI, I a just talking about the switch case here.

**Comment by nzakas** [comment]
Yeah, I'd expect `break` to be indented by default.

**Comment by gyandeeps** [comment]
I am going to re-write the whole ident rule and also fix this issue.
I already have it working.

**Comment by joeybaker** [comment]
Thank you @gyandeeps!

**Comment by gyandeeps** [comment]
u bet :+1:

**Comment by fourq** [comment]
This is still an issue with the latest version. screenshot http://bad-request.fourq.io/0y102l321D3p

**Comment by mgenware** [comment]
@fourq try this:
```js
"indent": [2, 2, {"SwitchCase": 1}]
```
[More info](http://eslint.org/docs/rules/indent.html)

**Comment by awitherow** [comment]
@mgenware that works for me as well, thanks for the hookup 👍

**Comment by xgqfrms-gildata** [comment]
`"indent": ["error", 4],`
